pkg/server/server.go

```diff
@@ -653,15 +653,26 @@ func (s *Server) ListUsers(
 	ctx context.Context,
 	req *openfgav1.ListUsersRequest,
 ) (*openfgav1.ListUsersResponse, error) {
+
 	typesys, err := s.typesystemResolver(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Slightly related to this PR because this code contains validation:

You shouldn't use `s.typesystemResolver, you should call `s.resolveTypesystem` because that is the one that returns actual server errors (with correct http status codes).

I.e. your code

```
curl -v --location 'localhost:8080/stores/01HTZAKTWJ5Y69NVC5TMRW7RGY/list-users' \
--data '{
    "authorization_model_id": "01HV4DX6NX1WNTFVDVAQW4WR58", // not found
    "object": {"type": "folder", "id": "5"},
    "relation": "can_read",
    "user_filters": [{"type": "user"}]
}'

< HTTP/1.1 500 Internal Server Error
{"code":"internal_error","message":"authorization model not found"}
```

versus ListObjects (for example)

```
curl -v --location 'localhost:8080/stores/01HTZAKTWJ5Y69NVC5TMRW7RGY/list-objects' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data '{
    "authorization_model_id": "01HV4DX6NX1WNTFVDVAQW4WR58", // not found
    "type": "document",
    "relation": "viewer",
    "user": "user:jon"
}'

< HTTP/1.1 400 Bad Request
{"code":"authorization_model_not_found","message":"Authorization Model '01HV4DX6NX1WNTFVDVAQW4WR58' not found"}
```

This is badly designed - at minimum, typesystemResolver should give us all the correct error codes :(